nts/implement Athlete Entity at Setup boundary

StoyanGoranov opened the pull request:
Closes: https://github.com/Not-Endurance/not-timing-system/issues/222

StoyanGoranov committed:
Modify Domain Entities

Add properties and expose public Create and Update method for Athlete

https://github.com/Not-Endurance/not-timing-system/issues/222

StoyanGoranov committed:
Create AthleteForm Blazor component

Includes AthleteFormModel.cs, AthleteFormUpdate.razor, and the update route added to
Endpoints.cs

https://github.com/Not-Endurance/not-timing-system/issues/222

StoyanGoranov committed:
Create AthleteRepository and add Athlete EntitySet to SetupState

Adding persistence of Athletes

https://github.com/Not-Endurance/not-timing-system/issues/222

StoyanGoranov committed:
Add NotSetList that uses the new AthleteForm component at EventPage

https://github.com/Not-Endurance/not-timing-system/issues/222

achobanov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Athletes/AthleteForm.razor:
```diff
@@ -0,0 +1,51 @@
+
+<EditForm Model="_model">
+    <MudTextField @ref="_nameField" @bind-Value="_model.Name" Label="@Localizer.Get("Name")" />
+    <MudTextField @bind-Value="_model.FeiId" Label="Fei Id" HelperText="@Localizer.Get("Leave empty if horse doesn't compete by FEI rules")" Variant="Variant.Text" Margin="Margin.Dense" />
```
_Leave empty if horse doesn't compete by FEI rules_ isn't exactly correct as it has nothing to do with the actual rules of the event. I think _Required for FEI export_ would be better.

Furthermore what happens if we decide to change that text later on. Isn't there a better approach if it's going to be repeated in multiple places?

achobanov replied:
What about my question?

achobanov commented on src-v2/Judge/NTS.Judge/Events/AthleteBehind.cs:
```diff
@@ -0,0 +1,37 @@
+    private readonly IRepository<Athlete> _athleteRepository;
+    private Athlete? _athlete;
+
+    public AthleteBehind(IRepository<Athlete> athleteRepository)
+    {
+        _athleteRepository = athleteRepository;
+    }
+    public Task<IEnumerable<Athlete>> GetAll()
+    {
+        return _athleteRepository.ReadAll();
+    }
+    public async Task<Athlete> Create(Athlete entity)
+    {
+        _athlete = await _athleteRepository.Create(entity);
+        return _athlete;
+    }
+
+    public async Task<Athlete> Update(Athlete entity)
```
Inconsistent empty lines

achobanov commented on src-v2/NTS.Persistance/Adapters/AthleteRepository.cs:
```diff
@@ -0,0 +1,12 @@
+using NTS.Domain.Setup.Entities;
+using NTS.Persistence.Setup;
+
+
```
Double empty line

achobanov reviewed (changes requested):
`closes #...` instead of related. Also you can work a bit on PR titles. A good PR title is a very short description of what it represents. This one is  a bit long. I'd change it to `Athlete entity in Setup`. Also a prefix for the target project is a good idea so `NTS/Athlete entity in Setup`

StoyanGoranov committed:
Address review 24.6.2024

https://github.com/Not-Endurance/not-timing-system/pull/246

achobanov reviewed (changes requested):
I've unresolved a previous comment.

StoyanGoranov committed:
Merge branch 'dev/nts' into nts/create-Athlete-Entity-Setup-boundary

Resolve conflicts

StoyanGoranov committed:
Add Text.cs with constants that contains UI messages

Apply the 'Optional_FEI_ID' message at AthleteForm.razor and
HorseForm.razor components

https://github.com/Not-Endurance/not-timing-system/pull/246

achobanov commented on src-v2/Judge/NTS.Judge.Blazor/Pages/Setup/Athletes/AthleteForm.razor:
```diff
@@ -5,11 +5,12 @@
 <EditForm Model="_model">
     <MudTextField @ref="_nameField" @bind-Value="_model.Name" Label="@Localizer.Get("Name")" />
-    <MudTextField @bind-Value="_model.FeiId" Label="Fei Id" HelperText="@Localizer.Get("Required for FEI export")" Variant="Variant.Text" Margin="Margin.Dense" />
+    <MudTextField @bind-Value="_model.FeiId" Label="Fei Id" HelperText="@Localizer.Get(Optional_FEI_ID)" Variant="Variant.Text" Margin="Margin.Dense" />
```
Lets keep constants UPPER_CASE.

StoyanGoranov committed:
Capitalize Text message

https://github.com/Not-Endurance/not-timing-system/issues/222
